### PR TITLE
Fix bogus host header in client

### PR DESCRIPTION
**Pull request by yakman2020**

### CODE REVIEW

**Comment by paulcallen** [comment]
need to add @Microsoft/omi-devs

File: Unix/http/httpclient.c [hunk 1]
```diff
@@ -2582,6 +2594,34 @@ MI_Result HttpClient_New_Connector2(
         self->connector->negoFlags    = FALSE;
         self->connector->hostname     = PAL_Strdup(host);
 
+        static const char HOST_HEADER[] = "Host: ";
+        int host_header_size_required = strlen(host)+sizeof(HOST_HEADER)+10; // 10 == max length of port plus CRLF
+
+        self->connector->hostHeader = PAL_Malloc(host_header_size_required);
```
Comment (paulcallen) [inline Unix/http/httpclient.c]: where is this getting free-ed?

File: Unix/http/httpclientauth.c [hunk 2]
```diff
@@ -1693,12 +1692,15 @@ HttpClient_NextAuthRequest(_In_ struct _HttpClient_SR_SocketData * self, _In_ co
     
         /* create header page */
 
-        char *request_header = PAL_Malloc(POST_HEADER_LEN+header_len+5);
+        char *request_header = PAL_Malloc(POST_HEADER_LEN+header_len+5+strlen(self->hostHeader));
```
Comment (paulcallen) [inline Unix/http/httpclientauth.c]: where does this get free-ed?

Comment (yakman2020) [inline Unix/http/httpclientauth.c]: In httpclient_Delete

File: Unix/http/httpclient.c (hunk 1 above)
Comment (yakman2020) [inline Unix/http/httpclient.c]: In httpclient_Delete

**Pull request merged**